### DISCUSSION TITLE
Setting up a Staging Server

### DISCUSSION THREAD

**Discussion by jangeador** [cookiecutter/cookiecutter-django#3403] [Q&A]
What is the proper way to setup a staging server using the production settings with docker? I am having to create a third file `staging.yml` which is identical to `production.yml` in everything except for the traefik container which builds from a different docker file in order to have a separate `traefik.yml` file with a different staging domain, also I am pointing this traefik.yml file to go to `caServer: https://acme-staging-v02.api.letsencrypt.org/directory` in order to have a semi-valid cert, not a real valid one. 

Please let me know if there is a better way to do this.

**Comment by taut-and-yare** [comment]
@jangeador Hello, could you share the files in question (staging.yml and Dockerfile) for reference?

**Comment by martibosch** [comment]
Hello!

in case it may interest anyone, I have designed an approach to set up a staging server in [the `staging` branch of my fork](https://github.com/martibosch/cookiecutter-django/commits/staging). It basically consists of four changes:
* adding a separate initial cookiecutter variable (i.e., `staging_domain_name`) for the staging domain name
* added staging environment variables for django (i.e., at `.envs/.staging/.django`) where `DJANGO_ALLOWED_HOSTS` is set to `staging_domain_name`. Optionally, you can change the AWS/GCP storage settings there too (e.g., if you use a dedicated S3 instance for the staging environment).
* added a `TRAEFIK_HOST` environment variable which takes the value of the production `domain_name` (in `.envs/.production/.traefik`) or the staging `staging_domain_name` (in a new file `.envs/.staging/.traefik`).
* finally, we add a `staging.yml` file for docker-compose which is based on the production environment but replaces the host/storage server addresses based on the environment variables described above. Then, you can use such a file as a configuration override, e.g., build the stack as in `docker-compose -f production.yml -f staging.yml build`.

It's just a first idea which may have drawbacks, but it's working so far and it avoids hard-copying (i.e., duplicating) the production settings for the staging server. Looking forward to recieve feedback. Best,
Martí
